### PR TITLE
Dawndawnwong/issue106

### PR DESCRIPTION
**Pull request by dawndawnwong**
- personal page ui
- connect my friend to backend
- connect my group name to backend
- connect my biography to backend

- add friend window

### CODE REVIEW

**Review by aprilemeraldchao — APPROVED** [review]
Looks good enough for now! I'll let you know if there's anything to change later

File: frontend/src/pages/PersonalPage.jsx [hunk 1]
```diff
@@ -1,12 +1,268 @@
...
+import UploadNotesWindow from "../components/uploadNotesWindow";
+import { Link } from "react-router-dom";
+import PDFthumbnail from "../components/PDFthumbnail";
 
 function PersonalPage(props) {
+    const [buttonAddFriend, setButtonAddFriend] = useState(false);
+    const [buttonGroup, setButtonGroup] = useState(false);
+    const [personalProfile, setButtonPersonalProfile] = useState(false);
+    const [friends, setFriends] = useState([]);
+    const [allGroups, setGroups] = useState([]);
```
Comment (aprilemeraldchao) [inline frontend/src/pages/PersonalPage.jsx:17]: For consistency, this should be ``[groups, setGroups]``

File: frontend/src/pages/PersonalPage.jsx [hunk 2]
```diff
@@ -1,12 +1,268 @@
...
+import AddFriendWindow from "../components/addFriendWindow";
+import "../css/personalPage.css";
+import UploadNotesWindow from "../components/uploadNotesWindow";
+import { Link } from "react-router-dom";
+import PDFthumbnail from "../components/PDFthumbnail";
 
 function PersonalPage(props) {
+    const [buttonAddFriend, setButtonAddFriend] = useState(false);
+    const [buttonGroup, setButtonGroup] = useState(false);
+    const [personalProfile, setButtonPersonalProfile] = useState(false);
```
Comment (aprilemeraldchao) [inline frontend/src/pages/PersonalPage.jsx:15]: For consistency, this should be ``buttonPersonalProfile, setButtonPersonalProfile``

File: frontend/src/pages/HomePage.jsx [hunk 3]
```diff
@@ -3,30 +3,50 @@ import SearchBar from "../components/searchBar";
...
+                                    title="SEARCH"
+                                    type="primary"
+                                    clicked={() => setUpdateSearch(true)}
+                                />
+                            </div>
+                        </div>
+                    </div>
+                }
+            />
+            <div>
```
Comment (aprilemeraldchao) [inline frontend/src/pages/HomePage.jsx:42]: Why is this is a separate div?

File: frontend/src/components/modalLeftWindow.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import "../css/personalPage.css"
+
+function ModalLeftWindow({body}) {
```
Comment (aprilemeraldchao) [inline frontend/src/components/modalLeftWindow.jsx:3]: This isn't really a modal window...it doesn't pop up and go on top of all the other page content...it's more like a panel

**Review by aprilemeraldchao — COMMENTED** [review]
image
Some general UI changes:
- maybe center the buttons in the left panel and shrink their text
- I think the edit profile button should be on the bottom of the panel
- the headers in the left panel should be smaller text
- the modal window for add friend isn't centered (see below)
image

File: frontend/src/components/modalLeftWindow.jsx (hunk 4 above)
Comment (dawndawnwong) [reply frontend/src/components/modalLeftWindow.jsx:3]: Changed to LeftPanel

File: frontend/src/pages/HomePage.jsx (hunk 3 above)
Comment (dawndawnwong) [reply frontend/src/pages/HomePage.jsx:42]: This is for blurring the entire background.

File: frontend/src/pages/PersonalPage.jsx (hunk 2 above)
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:15]: Changed it.

File: frontend/src/pages/PersonalPage.jsx (hunk 1 above)
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:17]: Fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 5]
```diff
@@ -1,12 +1,268 @@
...
+                        </div>
+
+                        <div className="d-flex right-side-middle">
+                            <ModalHeader title="My Notes" />
+                            <div className="d-flex flex-row right-side-down">
+                                <Button
+                                    title="BACK"
+                                    type="secondary"
+                                    url="/home"
+                                />
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:252]: Fixed it.

File: frontend/src/index.css [hunk 6]
```diff
@@ -50,6 +50,14 @@ html, body{
   font-size: 20px;
 }
 
+.agenda-big {
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/index.css:53]: Thank you, fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 7]
```diff
@@ -1,12 +1,268 @@
...
+        if (success) {
+            const userInfoResponseJSON = await userInfoResponse.json();
+            if (userInfoResponseJSON.username) {
+                setUser(userInfoResponseJSON.username);
+                setUserDescription(userInfoResponseJSON.description);
+            } else {
+                console.error("Could not get user information.");
+            }
+        } else {
+            return;
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:46]: Fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 8]
```diff
@@ -1,12 +1,268 @@
...
+        if (success) {
+            const friendResponseJSON = await friendsResponse.json();
+            setFriends([]);
+            if (friendResponseJSON.friends) {
+                for (const friend of friendResponseJSON.friends) {
+                    setFriends((arr) => [...arr, friend]);
+                }
+            }
+        } else {
+            return;
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:75]: Fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 9]
```diff
@@ -1,12 +1,268 @@
...
+
+        if (success) {
+            const groupResponseJSON = await groupsResponse.json();
+            setGroups([]);
+            if (groupResponseJSON.allGroups) {
+                for (const group of groupResponseJSON.allGroups) {
+                    setGroups((arr) => [...arr, group]);
+                }
+            } else {
+                setGroups("Mochi Group");
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:100]: Yes, deleted it.

File: frontend/src/pages/PersonalPage.jsx [hunk 10]
```diff
@@ -1,12 +1,268 @@
...
+            }
+        } else {
+            return;
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:103]: Fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 11]
```diff
@@ -1,12 +1,268 @@
...
         <>
             <Template
                 showSearch={true}
                 showProfile={true}
-                body={<h1>Personal Page</h1>}
+                body={
+                    <div className="d-flex flex-column left-side">
+                        <ModalHeader title={`Hi ${user}`} />
+                        <div className="d-flex flex-column align-items-left">
+                            <ModalLeftWindow
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:179]: Changed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 12]
```diff
@@ -1,12 +1,268 @@
...
+                                body={
+                                    <div className="d-flex flex row">
+                                        <ModalHeader title="My Friends" />
+                                        <div className="d-flex row agenda">
+                                            {friends.map((friend) => (
+                                                <h3>{friend.username}</h3>
+                                            ))}
+                                        </div>
+                                        <div className="d-flex">
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:188]: Yes, fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 13]
```diff
@@ -1,12 +1,268 @@
...
                 showSearch={true}
                 showProfile={true}
-                body={<h1>Personal Page</h1>}
+                body={
+                    <div className="d-flex flex-column left-side">
+                        <ModalHeader title={`Hi ${user}`} />
+                        <div className="d-flex flex-column align-items-left">
+                            <ModalLeftWindow
+                                body={
+                                    <div className="d-flex flex row">
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:181]: "d-flex flex row" changed to "flex-row".

File: frontend/src/pages/PersonalPage.jsx [hunk 14]
```diff
@@ -1,12 +1,268 @@
...
+                                                        color: "inherit",
+                                                        textDecoration:
+                                                            "inherit",
+                                                    }}
+                                                >
+                                                    <h3>{group.group_name}</h3>{" "}
+                                                </Link>
+                                            ))}
+                                        </div>
+                                        <div className="d-flex">
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:216]: Yes, fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 15]
```diff
@@ -1,12 +1,268 @@
...
+                                            <Button
+                                                title="CREATE/JOIN GROUP"
+                                                type="primary"
+                                                clicked={() =>
+                                                    setButtonGroup(true)
+                                                }
+                                            />
+                                        </div>
+                                        <div className="d-flex">
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:225]: Yes, fixed it.

File: frontend/src/pages/PersonalPage.jsx [hunk 16]
```diff
@@ -1,12 +1,268 @@
...
+                                                    )
+                                                }
+                                            />
+                                        </div>
+                                    </div>
+                                }
+                            />
+                        </div>
+                        <div className="d-flex flex-column right-side-top">
+                            <ModalHeader title="Biography" />
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/pages/PersonalPage.jsx:241]: Changed it.

File: frontend/src/components/signUpWindow.jsx [hunk 17]
```diff
@@ -65,8 +65,8 @@ function SignUpWindow(props) {
 
             const loginResponseJSON = await loginResponse.json();
             if (!loginResponseJSON.status_code) {
-                window.localStorage.setItem("authToken", responseJSON.token);
-                history.push("/home");
+                // window.localStorage.setItem("authToken", responseJSON.token);
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/components/signUpWindow.jsx:68]: Removed it.

File: frontend/src/components/addFriendWindow.jsx [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+                    "bearer " + window.localStorage.getItem("authToken"),
+            },
+            body: JSON.stringify({
+                username2,
+            }),
+        });
+
+        const responseJSON = await response.json();
+        if (responseJSON.username) {
+            alert("Succefully adding friend");
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/components/addFriendWindow.jsx:33]: Fixed it.

File: frontend/src/components/addFriendWindow.jsx [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+            history.push("/my_notes");
+        } else {
+            alert("Something went wrong with adding friend!");
```
[the remark this answers is not in the document]
Comment (dawndawnwong) [reply frontend/src/components/addFriendWindow.jsx:36]: I added one more alert for "There is no such username."

**Comment by aprilemeraldchao** [comment]
finished reviewing, any changes are now being mentioned in pr 194

**Pull request merged**